web/packages/teleterm/src/mainProcess/agentDownloader/agentDownloader.test.ts

```diff
@@ -22,17 +22,15 @@
 /* eslint jest/no-conditional-expect: 0 */
 
 import childProcess from 'node:child_process';
-import { PassThrough } from 'node:stream';
 import fs from 'node:fs';
 import fsPromises from 'node:fs/promises';
+import { PassThrough } from 'node:stream';
 import zlib from 'node:zlib';
-
 import tarFs from 'tar-fs';
```

Comment:
Node imports are grouped with 3rd-party libs.

This is what I found in the docs:

> Each import is matched against all regexes on the from string. The import ends up at the regex with the longest match. In case of a tie, the first matching regex wins.

https://github.com/lydell/eslint-plugin-simple-import-sort?tab=readme-ov-file#custom-grouping

I think the current groups might work if we get rid of `excludedPackagesPattern`? The default sort groups have this:

```
          // 3rd-party packages.

          // Things that start with a letter (or digit or underscore), or `@` followed by a letter.

          ['^@?\\w'],
```

…so I believe we have to make sure we match just on the first letter (but idk how easy that is when type imports come into play.

Reply:
Have you seen this example? https://github.com/lydell/eslint-plugin-simple-import-sort/blob/66d84f742959bf68f4575b6ac5a65978ceb7cfbc/examples/.eslintrc.js#L136-L148

It has type imports last but doesn't use negative lookahead.

Reply:
Assuming that we want to enable some kind of rule which would automatically pluck out type imports from regular imports, I wonder if putting type imports last within each group would make sense?

https://github.com/lydell/eslint-plugin-simple-import-sort/blob/66d84f742959bf68f4575b6ac5a65978ceb7cfbc/examples/.eslintrc.js#L210-L228

Reply:
At the moment, no matter which option we pick wrt type imports isn't going to seem like a big deal. But down the road, assuming that their usage will be more common in the codebase, I wonder if it wouldn't look more tidy if type imports were within individual groups rather their own group.

Personally, when I look at imports, the most important information for me is whether the package is from the stdlib, or a 3rd-party lib, or maybe one of our packages. From that perspective, it'd make sense for type imports to be put within each particular group as well.

Reply:
Gotcha, that makes sense to me. Will take another look at this later, and see how things look grouping type imports at the end of groups.

Part of the issue we have currently is allowing importing types/interfaces alongside functions/enums/consts; I think enforcing using `import type` syntax would help clean things up as well.

Reply:
I figured out how to put type imports at the end of each group in https://github.com/gravitational/teleport/pull/50619 (left a comment in the ESLint config explaining what's going on) and credited you in the commits under both OSS and ent PRs. I also based this on our updated ESLint 9 config.